internal/services/signalr/web_pubsub_socketio_resource.go

```diff
@@ -0,0 +1,619 @@
+				if err != nil {
+					return fmt.Errorf("flattening `identity`: %+v", err)
+				}
+				state.Identity = pointer.From(flattenedIdentity)
+
+				if sku := model.Sku; sku != nil {
+					state.Sku = flattenSkuToModel(sku)
+				}
+
+				if props := model.Properties; props != nil {
+					liveTrace := flattenLiveTraceConfigToMap(props.LiveTraceConfiguration)
+
+					state.AADAuthEnabled = !pointer.From(props.DisableAadAuth)
+					state.ExternalIP = pointer.From(props.ExternalIP)
+					state.HostName = pointer.From(props.HostName)
+					state.PublicPort = pointer.From(props.PublicPort)
+					state.ServerPort = pointer.From(props.ServerPort)
+
+					state.LiveTraceEnabled = liveTrace["enabled"]
+					state.LiveTraceConnectivityLogsEnabled = liveTrace["connectivityLogsEnabled"]
+					state.LiveTraceHttpRequestLogsEnabled = liveTrace["httpLogsEnabled"]
+					state.LiveTraceMessagingLogsEnabled = liveTrace["messagingLogsEnabled"]
+
+					state.LocalAuthEnabled = !pointer.From(props.DisableLocalAuth)
+					state.PublicNetworkAccess = pointer.From(props.PublicNetworkAccess)
+
+					if socketio := props.SocketIO; socketio != nil {
+						state.ServiceMode = pointer.From(socketio.ServiceMode)
+					}
+
+					if tls := props.Tls; tls != nil {
+						state.TlsClientCertEnabled = pointer.From(tls.ClientCertEnabled)
+					}
+				}
+			}
+
+			if keyModel := keys.Model; keyModel != nil {
+				state.PrimaryAccessKey = pointer.From(keyModel.PrimaryKey)
+				state.PrimaryConnectionString = pointer.From(keyModel.PrimaryConnectionString)
+				state.SecondaryAccessKey = pointer.From(keyModel.SecondaryKey)
+				state.SecondaryConnectionString = pointer.From(keyModel.SecondaryConnectionString)
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (w WebPubSubSocketIOResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.SignalR.WebPubSubClient.WebPubSub
+
+			id, err := webpubsub.ParseWebPubSubID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config WebPubSubSocketIOResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
+			if existing.Model.Sku == nil {
+				return fmt.Errorf("retrieving %s: `model.Sku` was nil", *id)
+			}
+
+			payload := webpubsub.WebPubSubResource{
+				Location:   location.Normalize(config.Location),
+				Properties: &webpubsub.WebPubSubProperties{},
+				Sku:        existing.Model.Sku,
+			}
+			props := payload.Properties
+
+			rd := metadata.ResourceData
+			if rd.HasChange("sku") {
+				payload.Sku = expandSkuFromModel(config.Sku)
```

Comment:
We should be a little more verbose here in what sku we're expanding/flattening. If we add another resource that has a sku, we might not be able to use this method again so by being more verbose, we can easily add in another methods that expand skus for different resources without them clashing over each other

```suggestion
				payload.Sku = expandWebPubSubSocketIOSkuFromModel(config.Sku)
```